### PR TITLE
fix(javascript-sdk): capitalize headers

### PR DESCRIPTION
**Pull request by cerebrl**
# JIRA Ticket

https://bugster.forgerock.org/jira/browse/SDKS-2141

# Description

# Type of Change

Please Delete options that are not relevant

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manually ran tests, but the browser developer tools (Chrome) seem to auto-lowercase all headers. Firefox seems to auto-lowercase only some headers. I'm not sure how to perfectly valid this.

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [x] All tasks listed in the user story have been completed.
- [x] Coded to standards.
- [ ] Code peer-reviewed.
- [x] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [x] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

**Review by ryanbas21 — DISMISSED** [review]
Can you add the changelog to the one in `package/javascript-sdk/changelog` that is the one we include in the publish

File: packages/javascript-sdk/src/session-manager/index.ts [hunk 1]
```diff
@@ -28,8 +28,8 @@ abstract class SessionManager {
     const init: RequestInit = {
       credentials: 'include',
       headers: new Headers({
-        'accept-api-version': 'protocol=1.0,resource=2.0',
-        'x-requested-with': REQUESTED_WITH,
+        'Accept-Api-Version': 'protocol=1.0,resource=2.0',
```
Comment (daveadams56) [inline packages/javascript-sdk/src/session-manager/index.ts:31]: Is "Accept-Api-Version" the correct casing for this header? In the ticket Brandon mentions "Accept-API-Version", so I just wanted to get confirmation which one is correct

File: packages/javascript-sdk/src/http-client/helpers.ts [hunk 2]
```diff
@@ -31,13 +31,13 @@ export function addAuthzInfoToHeaders(
   const headers = new Headers(init.headers);
 
   if (advices.AuthenticateToServiceConditionAdvice) {
-    headers.set('x-tree', advices.AuthenticateToServiceConditionAdvice[0]);
+    headers.set('X-Tree', advices.AuthenticateToServiceConditionAdvice[0]);
   } else if (advices.TransactionConditionAdvice) {
-    headers.set('x-txid', advices.TransactionConditionAdvice[0]);
+    headers.set('X-Txid', advices.TransactionConditionAdvice[0]);
```
Comment (daveadams56) [inline packages/javascript-sdk/src/http-client/helpers.ts:36]: I'm getting nothing useful from backstage searching for "X-Txid" or even "Txid", so I just wanted to ask a similar questions as "Accept-Api-Version" below - is our strategy here just to capitalise the first letter of each word here?

File: packages/javascript-sdk/src/http-client/helpers.ts [hunk 3]
```diff
@@ -31,13 +31,13 @@ export function addAuthzInfoToHeaders(
   const headers = new Headers(init.headers);
 
   if (advices.AuthenticateToServiceConditionAdvice) {
-    headers.set('x-tree', advices.AuthenticateToServiceConditionAdvice[0]);
+    headers.set('X-Tree', advices.AuthenticateToServiceConditionAdvice[0]);
   } else if (advices.TransactionConditionAdvice) {
-    headers.set('x-txid', advices.TransactionConditionAdvice[0]);
```
Comment (daveadams56) [inline packages/javascript-sdk/src/http-client/helpers.ts:36]: I can see in our mock API we refer to "x-txid" e.g. [routes.resource.js](https://github.com/ForgeRock/forgerock-javascript-sdk/blob/fix_sdk-headers_SDKS-2141/e2e/mock-api/src/app/routes.resource.js#L31) - should we also update the header name here?

**Review by daveadams56 — COMMENTED** [review]
Looks good, but I would want to clarify the casing of some of the headers that I've left comments on, and decide either way about the mock API. Happy to approve once we discuss those two.

File: packages/javascript-sdk/src/session-manager/index.ts (hunk 1 above)
Comment (cerebrl) [reply packages/javascript-sdk/src/session-manager/index.ts:31]: I can make the change to reflect the request in the ticket.

File: packages/javascript-sdk/src/http-client/helpers.ts (hunk 3 above)
Comment (cerebrl) [reply packages/javascript-sdk/src/http-client/helpers.ts:36]: Functionally, the official spec says headers are case insensitive. I can make that change, but I don't believe it's necessary.

File: packages/javascript-sdk/src/http-client/helpers.ts (hunk 2 above)
Comment (cerebrl) [reply packages/javascript-sdk/src/http-client/helpers.ts:36]: Anything that starts with `x-` is a customer header, so there's no right or wrong way to do this. I can make a change to be a bit more consistent at least internally.

**Pull request merged**